lib/src/open_api/spec.dart

```diff
@@ -67,6 +72,10 @@ class OpenApi with _$OpenApi {
     /// this definition. To make security optional, an empty security requirement ({})
     /// can be included in the array.
     List<Security>? security,
+
+    /// A mapping of any extra schemas that this generator created and the parent schema
+    /// that they were created from. This is used to improve the generated schema library
+    @Default({}) Map<String, List<String>> extraSchemaMapping,
```

Comment:
New property added to  `OpenApi` object to track extra schema(s) that were generated and from which parent they belong. Used in the generators, but exposed to the user since `openapi_spec` is making assumptions and essentially generating new schemas on the fly that may not be obvious from inspecting the source spec.